compiler/passes/scopeResolve.cpp

```diff
@@ -344,7 +344,10 @@ static void processImportExprs() {
       // A ModuleSymbol's modUseList is intended to be for initializing modules
       // in an appropriate order.  If we're using an enum symbol, we don't
       // need to worry about this.
-      enclosingModule->moduleUseAdd(mod);
+
+      // Avoid a self-reference in the mod-use list
+      if (mod != enclosingModule)
+        enclosingModule->moduleUseAdd(mod);
```

Comment:
Hmm, I'm thinking: would it better to update the moduleUseAdd method to silently drop attempts to add itself on the floor instead of modifying the code in two places?  I think these are the only places it is called, so it probably doesn't matter too much.